homeassistant/components/ambiclimate/climate.py

```diff
@@ -137,100 +137,50 @@ async def set_temperature_mode(service):
 class AmbiclimateEntity(ClimateEntity):
     """Representation of a Ambiclimate Thermostat device."""
 
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_target_temperature_step = 1
+    _attr_supported_features = SUPPORT_FLAGS
+    _attr_hvac_modes = [HVAC_MODE_HEAT, HVAC_MODE_OFF]
+
     def __init__(self, heater, store):
         """Initialize the thermostat."""
         self._heater = heater
         self._store = store
         self._data = {}
-
-    @property
-    def unique_id(self):
-        """Return a unique ID."""
-        return self._heater.device_id
-
-    @property
-    def name(self):
-        """Return the name of the entity."""
-        return self._heater.name
-
-    @property
-    def device_info(self):
-        """Return the device info."""
-        return {
+        self._attr_unique_id = self._heater.device_id
+        self._attr_name = self._heater.name
+        self._attr_device_info = {
             "identifiers": {(DOMAIN, self.unique_id)},
             "name": self.name,
             "manufacturer": "Ambiclimate",
         }
-
-    @property
-    def temperature_unit(self):
-        """Return the unit of measurement which this thermostat uses."""
-        return TEMP_CELSIUS
-
-    @property
-    def target_temperature(self):
-        """Return the target temperature."""
-        return self._data.get("target_temperature")
-
-    @property
-    def target_temperature_step(self):
-        """Return the supported step of target temperature."""
-        return 1
-
-    @property
-    def current_temperature(self):
-        """Return the current temperature."""
-        return self._data.get("temperature")
-
-    @property
-    def current_humidity(self):
-        """Return the current humidity."""
-        return self._data.get("humidity")
-
-    @property
-    def min_temp(self):
-        """Return the minimum temperature."""
-        return self._heater.get_min_temp()
-
-    @property
-    def max_temp(self):
-        """Return the maximum temperature."""
-        return self._heater.get_max_temp()
-
-    @property
-    def supported_features(self):
-        """Return the list of supported features."""
-        return SUPPORT_FLAGS
-
-    @property
-    def hvac_modes(self):
-        """Return the list of available hvac operation modes."""
-        return [HVAC_MODE_HEAT, HVAC_MODE_OFF]
-
-    @property
-    def hvac_mode(self):
-        """Return current operation."""
-        if self._data.get("power", "").lower() == "on":
-            return HVAC_MODE_HEAT
-
-        return HVAC_MODE_OFF
-
-    async def async_set_temperature(self, **kwargs):
+        self._attr_target_temperature = self._data.get("target_temperature")
+        self._attr_current_temperature = self._data.get("temperature")
+        self._attr_current_humidity = self._data.get("humidity")
```

Comment:
This should either stay as properties or set after:
https://github.com/home-assistant/core/blob/095e839db7452844e233a336567437904fc688fb/homeassistant/components/ambiclimate/climate.py#L194
If you set them in init they won't update when they change.